src/Chapter2.hs

```diff
@@ -841,7 +850,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n l = 
+  let llen = length l
+      ls = take (2 * llen) (cycle l)
```

Comment:
I think you don't need to take `2 * llen` elements first. `cycle` returns an infinite list, so if you drop from it and take, then you'll have a finite list in the end